modin/pandas/test/test_io.py

```diff
@@ -1102,1229 +1054,1120 @@ def test_read_csv_internal(
-        modin_df = pd.read_json(data)
-    # For I/O objects we need to rewind to reuse the same object.
-    if hasattr(data, "seek"):
-        data.seek(0)
-    df_equals(modin_df, pandas.read_json(data))
+        pandas_df = wrapped_read_table(unique_filename, method="pandas")
+        modin_df = wrapped_read_table(unique_filename, method="modin")
 
+        df_equals(modin_df, pandas_df)
 
-def test_from_html():
-    setup_html_file(NROWS)
 
-    pandas_df = pandas.read_html(TEST_HTML_FILENAME)[0]
-    modin_df = pd.read_html(TEST_HTML_FILENAME)
+class TestParquet:
+    def test_read_parquet(self, make_parquet_file):
+        unique_filename = get_unique_filename(extension="parquet")
+        make_parquet_file(filename=unique_filename)
 
-    df_equals(modin_df, pandas_df)
+        eval_io(
+            fn_name="read_parquet",
+            # read_parquet kwargs
+            path=unique_filename,
+        )
 
-    teardown_html_file()
+    def test_read_parquet_with_columns(self, make_parquet_file):
+        unique_filename = get_unique_filename(extension="parquet")
+        make_parquet_file(filename=unique_filename)
 
+        eval_io(
+            fn_name="read_parquet",
+            # read_parquet kwargs
+            path=unique_filename,
+            columns=["col1"],
+        )
 
-@pytest.mark.skip(reason="No clipboard on Travis")
-def test_from_clipboard():
-    setup_clipboard(NROWS)
+    def test_read_parquet_partition(self, make_parquet_file):
 
-    pandas_df = pandas.read_clipboard()
-    modin_df = pd.read_clipboard()
+        unique_filename = get_unique_filename(extension=None)
+        make_parquet_file(filename=unique_filename, directory=True)
+        eval_io(
+            fn_name="read_parquet",
+            # read_parquet kwargs
+            path=unique_filename,
+        )
 
-    df_equals(modin_df, pandas_df)
+    def test_read_parquet_partition_with_columns(self, make_parquet_file):
 
+        unique_filename = get_unique_filename(extension=None)
+        make_parquet_file(filename=unique_filename, directory=True)
+        eval_io(
+            fn_name="read_parquet",
+            # read_parquet kwargs
+            path=unique_filename,
+            columns=["col1"],
+        )
 
-@pytest.mark.xfail(reason="read_excel is broken for now, see #1733 for details")
-@check_file_leaks
-def test_from_excel():
-    setup_excel_file(NROWS)
+    def test_read_parquet_partitioned_columns(self, make_parquet_file):
 
-    pandas_df = pandas.read_excel(TEST_EXCEL_FILENAME)
-    modin_df = pd.read_excel(TEST_EXCEL_FILENAME)
+        unique_filename = get_unique_filename(extension=None)
+        make_parquet_file(filename=unique_filename, partitioned_columns=["col1"])
+        eval_io(
+            fn_name="read_parquet",
+            # read_parquet kwargs
+            path=unique_filename,
+        )
 
-    df_equals(modin_df, pandas_df)
+    def test_read_parquet_partitioned_columns_with_columns(self, make_parquet_file):
```

Comment:
It seems to me that `test_read_parquet_partitioned_columns_with_columns` can be easy merged with `test_read_parquet_partitioned_columns`.

Reply:
Ok, tests are merged.